mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -73,201 +75,182 @@ def space_adapter(self) -> Optional[BaseSpaceAdapter]:
     def register(
         self,
         *,
-        configs: pd.DataFrame,
-        scores: pd.DataFrame,
-        context: Optional[pd.DataFrame] = None,
-        metadata: Optional[pd.DataFrame] = None,
+        observations: Optional[Observations] = None,
+        observation: Optional[Observation] = None,
     ) -> None:
         """
-        Wrapper method, which employs the space adapter (if any), before registering the
-        configs and scores.
+        Register all observations at once. Exactly one of observations or observation
+        must be provided.
 
         Parameters
         ----------
-        configs : pd.DataFrame
-            Dataframe of configs / parameters. The columns are parameter names and
-            the rows are the configs.
-        scores : pd.DataFrame
-            Scores from running the configs. The index is the same as the index of the configs.
+        observations: Observations
+            The observations to register.
+        observation: Observation
+            The observation to register.
+        """
+        assert (observations is None and observation is not None) or (
+            observations is not None and observation is None
+        ), "Only one of observations or observation can be provided."
 
-        context : pd.DataFrame
-            Not Yet Implemented.
+        if observation is not None:
+            self.register_single(observation=observation)
+        if observations is not None:
+            for obs in observations.to_list():
+                self.register_single(observation=obs)
```

Comment:
Whether this is a for loop over singleton registations or some other method of say rebuilding the internal model in bulk should be an implementation detail left up to the various backends, not done for all Optimizers in the wrapper class.